Sources/BuildSystemIntegration/SwiftPMBuildSystem.swift

```diff
@@ -565,6 +565,27 @@ package actor SwiftPMBuildSystem: BuiltInBuildSystem {
           data: SourceKitSourceItemData(isHeader: true).encodeToLSPAny()
         )
       }
+      sources += swiftPMTarget.resources.map {
+        SourceItem(
+          uri: DocumentURI($0),
+          kind: $0.isDirectory ? .directory : .file,
+          generated: false,
+        )
+      }
+      sources += swiftPMTarget.ignored.map {
+        SourceItem(
+          uri: DocumentURI($0),
+          kind: $0.isDirectory ? .directory : .file,
+          generated: false,
+        )
+      }
+      sources += swiftPMTarget.others.map {
+        SourceItem(
+          uri: DocumentURI($0),
+          kind: $0.isDirectory ? .directory : .file,
+          generated: false,
+        )
+      }
```

Comment:
The maps are all the same. Could we have only one copy of them as follows?

```suggestion
      sources += (swiftPMTarget.resources + swiftPMTarget.ignored + swiftPMTarget.others).map {
        SourceItem(
          uri: DocumentURI($0),
          kind: $0.isDirectory ? .directory : .file,
          generated: false,
        )
      }
```

Reply:
Good call!